services/openvpn/client_config.go

```diff
@@ -100,9 +100,11 @@ func NewClientConfigFromSession(sessionConfig []byte, configDir string, runtimeD
 	if vpnConfig.LocalPort > 0 {
 		clientFileConfig.OriginalRemoteIP = vpnConfig.RemoteIP
 		clientFileConfig.OriginalRemotePort = vpnConfig.RemotePort
-		vpnConfig.RemoteIP = "127.0.0.1"
-		// TODO: randomize this too?
-		vpnConfig.RemotePort = vpnConfig.LocalPort + 1
+		if isMobile {
+			vpnConfig.RemoteIP = "127.0.0.1"
```

Comment:
DI should inject these differences instead of conditional programming. Especially inside `openvpn` package, this does not know about NAT Proxy at all

Reply:
We are in constructor here, not sure how we can avoid conditional here, we can have 2 different constructors then? Is it better?

Reply:
Make 2 different constructions of `vpnConfig `